src/picotorrent/core/configuration.cpp

```diff
@@ -214,3 +216,30 @@ void Configuration::UpsertListenInterface(Configuration::ListenInterface const&
         stmt->Execute();
     }
 }
+
+bool Configuration::IsSystemDarkMode()
+{
+    bool systemUsesDarkTheme = false;
+    HKEY hKey = 0;
+    DWORD dwValue = 1;
+    DWORD dwBufSize = sizeof(dwValue);
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, TEXT("Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize"), 0, KEY_QUERY_VALUE, &hKey) == ERROR_SUCCESS)
+    {
+        RegQueryValueEx(hKey, TEXT("AppsUseLightTheme"), NULL, NULL, (LPBYTE)&dwValue, &dwBufSize);
+    }
+    if (dwValue == 0)
+    {
+        systemUsesDarkTheme = true;
+    }
+    return systemUsesDarkTheme;
+}
+
+bool Configuration::IsDarkMode()
+{
+    auto val = Configuration::Get<std::string>("theme_id").value_or("system");
```

Comment:
```suggestion
    return Configuration::Get<std::string>("theme_id").value_or("system") == "light"
        ? false
        : IsSystemDarkMode();
```

This can probably be simplified to the above.